src/python/qgspythonutilsimpl.cpp

```diff
@@ -169,8 +178,7 @@ void QgsPythonUtilsImpl::initPython( QgisInterface* interface )
   // initialize 'iface' object
   runString( "qgis.utils.initInterface(" + QString::number(( unsigned long ) interface ) + ")" );
 
-  QString startuppath = homePythonPath() + " + \"/startup.py\"";
-  runString( "if os.path.exists(" + startuppath + "): from startup import *\n" );
+  runString( QString("qgis.user.startup(%1)").arg( homePythonPath() ) );
```

Comment:
@wonder-sk new qgis.user module is called from here to kick off the Python user startup process.